internal/generic/generic.go

```diff
@@ -48,3 +51,94 @@ func CtxEither[left, right any](ctx context.Context,
 	}()
 	return eithers
 }
+
+// CtxBoth receives a single value from both channels
+// and binds them as a Couple,
+// until either channel closes or the context is done.
+func CtxBoth[both Couple[left, right], left, right any](ctx context.Context,
```

Comment:
TODO: we'll likely want to use this style `[both Couple[left, right], left, right any]` for CtxEither as well.
It makes the signature gross but it seems more correct.
We're defining a type `both` for this scope, and we do use it inside the body and return.